dotnet/src/Agents/Core/ChatCompletionAgent.cs

```diff
@@ -34,14 +34,25 @@ public sealed class ChatCompletionAgent : ChatHistoryKernelAgent
         }
         chat.AddRange(history);
 
-        var messages =
+        int messageCount = chat.Count;
+
+        IReadOnlyList<ChatMessageContent> messages =
             await chatCompletionService.GetChatMessageContentsAsync(
                 chat,
                 this.ExecutionSettings,
                 this.Kernel,
                 cancellationToken).ConfigureAwait(false);
 
-        foreach (var message in messages ?? [])
+        for (int messageIndex = messageCount; messageIndex < chat.Count; messageIndex++)
+        {
+            ChatMessageContent message = chat[messageIndex];
+
+            message.AuthorName = this.Name;
+
+            yield return message;
+        }
```

Comment:
Won't returning the entire chat history by this method lead to duplicates in the chat history stored in the ChatHistoryChannel class?
<img width="815" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/b742f3a8-9395-4453-9410-546effea2f78">

Reply:
I believe only the incremental messages are added.

Reply:
Right